src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,47 @@
 import React from 'react';
+import { Post } from '../../types/post';
+import { UserInfo } from '../UserInfo';
+import { CommentList } from '../CommentList';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+type Props = {
+  post: Post;
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const {
+    title,
+    body,
+    user,
+    comments,
+  } = post;
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{title}</h3>
+
+        <p>
+          {' Posted by  '}
+
+          {user && (
+            <UserInfo user={user} />
+          )}
+        </p>
+      </div>
+
+      <p className="PostInfo__body">
+        {body}
+      </p>
+
+      <hr />
+
+      {!comments.length && (
```

Comment:
Better to compare explicitly, [check here why](https://mate-academy.github.io/style-guides/javascript.html#153use-shortcuts-for-booleans-but-explicit-comparisons-for-strings-and-numbers)
```suggestion
      {comments.length > 0 && (
```